src/datreant/core/agglimbs.py

```diff
@@ -245,39 +316,96 @@ def remove(self, *categories):
         Any number of categories (keys) can be given as arguments, and these
         keys (with their values) will be deleted.
 
-        :Arguments:
-            *categories*
-                Categories to delete.
-
+        Parameters
+        ----------
+        *categories
+            Categories to delete.
         """
-        with self._treant._write:
-            for key in categories:
-                # continue even if key not already present
-                self._treant._state['categories'].pop(key, None)
+        for member in self._collection:
+            member.categories.remove(*categories)
 
     def clear(self):
         """Remove all categories from Treant.
 
         """
-        with self._treant._write:
-            self._treant._state['categories'] = dict()
+        for member in self._collection:
+            member.categories.clear()
 
     def keys(self):
-        """Get category keys.
-
-        :Returns:
-            *keys*
-                keys present among categories
+        """Get the unique Categories (keys) of all Treants in collection.
+
+        Returns
+        -------
+        list of list
+            Keys for each member are returned as a list, where the list of the
+            lists for each member has the same order as the members in the
+            collection.
         """
-        with self._treant._read:
-            return self._treant._state['categories'].keys()
+        return [member.categories.keys() for member in self._collection]
 
     def values(self):
-        """Get category values.
-
-        :Returns:
-            *values*
-                values present among categories
+        """Get the unique category values of all Treants in collection.
+
+        Returns
+        -------
+        list of list
+            Values for each member are returned as a list, where the list of
+            the lists for each member has the same order as the members in the
+            collection.
         """
-        with self._treant._read:
-            return self._treant._state['categories'].values()
+        return [member.categories.values() for member in self._collection]
+
+    def groupby(self, keys):
+        """Return groupings of Treants based on Categories.
```

Comment:
`groupby` should work differently than this. It should be analogous to doing a groupby on a column in a pandas DataFrame, in which you essentially get back a DataFrame for each combination of values observed in the specified columns (keys).

What `groupby` should do in the simplest case is take a `key` as input and return a dict of `Bundle`s, with the keys of this output dict being the values for the input key and the values being `Bundle`s of `Treant`s that had those values for the input key. `Treant`s that don't have the input key at all can be left out.

The more complex case is that `groupby` should also be able to take a list of keys and give back an output dict with all combinations of values (as tuples) for the input keys found for those keys as keys, with values `Bundle`s of `Treant`s with those combinations of values as members.

There might be better ways of structuring the outputs, but something like this would be a proper `groupby`. Thoughts?